src/combinatorics/quiver.lean

```diff
@@ -0,0 +1,89 @@
+/-
+Copyright (c) 2021 David Wärn. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Wärn
+-/
+import tactic
+/-!
+# Quivers
+
+This module defines quivers. A quiver `G` on a type `V` of vertices assigns to every
+pair `a b : V` of vertices a type `G a b`, thought of as edges from `a` to `b`. This
+is a very permissive notion of graph.
+-/
+
+-- We use the same universe order as in category theory.
+universes v u
+
+/-- A quiver `G` on a type `V` of vertices assigns to every pair `a b : V` of vertices
+    a type `G a b`, thought of as edges from `a` to `b`. -/
+def quiver (V : Type u) := V → V → Type v
+
+/-- The default quiver on `V` is the empty quiver. -/
+instance {V} : inhabited (quiver.{v} V) :=
+⟨λ a b, pempty⟩
```

Comment:
It might be useful defining `quiver.empty` and then using this to implement the instance.  (It seems like no one uses `inhabited` instances for what their value is, but I could be wrong.)